### ISSUE TITLE
v3.0.0-rc.2 decodeUriComponent is called before route match, breaking links with urlencoded slash (%2F)

### ISSUE THREAD

**Issue by justicecurcian** [molefrog/wouter#402]
```javascriptreact
<Route path="/:foo"><Foo></Route>
<Link href={'/' + encodeUriComponent('foo/bar')}>Go to Foo</Link>
```

`decodeUriComponent` is called before route match, so the url matcher thinks slashes (%2F) in params are path separators

**Comment by molefrog** [comment]
Yep, that's related to #400, the fix is under review.

**Comment by molefrog** [comment]
Fixed in v3.0.0-rc.3
